components/microfrontend/pkg/apis/ui/v1alpha1/types.go

```diff
@@ -0,0 +1,75 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// +genclient
+// +genclient:noStatus
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type MicroFrontend struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+	Spec              MicroFrontendSpec `json:"spec"`
+}
+
+// +genclient
+// +genclient:noStatus
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type ClusterMicroFrontend struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+	Spec              ClusterMicroFrontendSpec `json:"spec"`
+}
+
+func (rem *MicroFrontend) GetObjectKind() schema.ObjectKind {
+	return &MicroFrontend{}
+}
+
+func (rem *ClusterMicroFrontend) GetObjectKind() schema.ObjectKind {
+	return &ClusterMicroFrontend{}
+}
+
+type MicroFrontendSpec struct {
+	Version         string           `json:"version"`
```

Comment:
As a part of spec is shared between MF and CMF, I would suggest to create a type `CommonMicroFrontendSpec`. It would be inlined in `MicroFrontendSpec` and `ClusterMicroFrontendSpec`.